### DISCUSSION TITLE
Mirroring repos into a user defined category

### DISCUSSION THREAD

**Discussion by andrew-tram** [eclipse-tycho/tycho#4422] [Q&A]
I'm mirroring some Eclipse projects repositories to host internally using `tycho-p2-extras-plugin:mirror`.  

And for example, I have these repos as my source,

```
    <repository>
	    <url>https://download.eclipse.org/webtools/downloads/drops/R3.35.0/R-3.35.0-20240825085431/repository/</url>
	    <layout>p2</layout>
    </repository>
    <repository>
	    <url>https://download.eclipse.org/tools/orbit/simrel/orbit-aggregation/release/4.33.0</url>
	    <layout>p2</layout>
    </repository>
    <repository>
	    <url>https://download.eclipse.org/datatools/updates/release/1.16.1/</url>
	    <layout>p2</layout>
    </repository>
    <repository>
	    <url>https://download.eclipse.org/mpc/drops/1.11.0/v20240709-1650/</url>
	    <layout>p2</layout>
    </repository>
    <repository>
	    <url>https://download.eclipse.org/technology/m2e/releases/2.6.2/</url>
	    <layout>p2</layout>
    </repository>
    <repository>
	    <url>https://download.eclipse.org/buildship/updates/e431/releases/3.x/3.1.10.v20240802-1211/</url>
	    <layout>p2</layout>
    </repository>
    <repository>
	    <url>https://download.eclipse.org/tools/orbit/simrel/maven-jetty/release/12.0.14</url>
	    <layout>p2</layout>
    </repository>
```

Is there a way to not use the categories defined in the mirrored repositories and either add them into my own parent category like, "Eclipse Projects" or just have all the content in my defined category without having to spell it all out in a category.xml?

![image](https://github.com/user-attachments/assets/e3a5e2c5-b127-42c9-a5c9-b9d5abfc2ded)

**Comment by laeubi** [comment]
You maybe want to try out https://github.com/eclipse-tycho/tycho/blob/main/RELEASE_NOTES.md#new-mirror-target-platform-mojo I'm not completely sure it mirrors categories as well, but we can enhance it.

Beside that there is already something like https://tycho.eclipseprojects.io/doc/latest/tycho-extras/tycho-p2-extras-plugin/mirror-mojo.html#includeFeatures so we could maybe even have an `includeCategories` option.

**Reply by andrew-tram** [reply]
Yeah, I wasnt sure what to make of the new `mirror-target-platform` mojo. 

But with the `extras-plugin` I think that option or having something like `parentCategoryName` would be good and categories already defined from the source repo will go under the parent

```
        <property name='org.eclipse.equinox.p2.name' value='Buildship Extras - Incubating'/>
        <property name='org.eclipse.equinox.p2.type.category' value='true'/>
```
